bitnami/postgresql/templates/statefulset.yaml

```diff
@@ -189,7 +189,7 @@ spec:
             - name: POSTGRES_CLUSTER_APP_NAME
               value: {{ .Values.replication.applicationName }}
             {{- end }}
-            {{- if and .Values.postgresqlPostgresPassword (not (eq .Values.postgresqlUsername "postgres")) }}
+            {{- if and (or (not (include "postgresql.createSecret" .)) .Values.postgresqlPostgresPassword) (not (eq .Values.postgresqlUsername "postgres")) }}
```

Comment:
These conditions are really hard to read (and understand):

```lua
{{- if and (or (not (include "postgresql.createSecret" .)) .Values.postgresqlPostgresPassword) (not (eq .Values.postgresqlUsername "postgres")) }}
```

"If a new secret does not need to be created or a password is specified, then also check that the username is Postgres"

I think that just by re-ordering this would make it simpler:

```lua
{{- if and (not (eq .Values.postgresqlUsername "postgres")) (or .Values.postgresqlPostgresPassword (not (include "postgresql.createSecret" .))) }}
```

"When the username is postgres, and either a password was specified or a new secret does not need to be created"

I think we could simplify it even more if we improve the `postgresql.createSecret` helper:

```lua
{{/*
Return true if a secret object should be created
*/}}
{{- define "postgresql.useExistingSecret" -}}
{{- if or .Values.global.postgresql.existingSecret .Values.existingSecret -}}
    {{- true -}}
{{- end -}}
{{- end -}}

{{/*
Return true if a secret object should be created
*/}}
{{- define "postgresql.createSecret" -}}
{{- if not (include "postgresql.useExistingSecret" .) -}}
    {{- true -}}
{{- end -}}
{{- end -}}
```

And then:

```lua
{{- if and (not (eq .Values.postgresqlUsername "postgres")) (or .Values.postgresqlPostgresPassword (include "postgresql.useExistingSecret" .)) }}
```

"When the username is postgres, and either a password was specified or an existing secret will be used"

Reply:
Thanks very much for your suggestions, changes have been applied 😁